kubernetes/cmsweb/scripts/deploy-secrets.sh

```diff
@@ -1,24 +1,31 @@
 #!/bin/bash
 # helper script to deploy given service with given tag to k8s infrastructure
 
-if [ $# -ne 3 ]; then
-    echo "Usage: deploy-secrets.sh <namespace> <service-name> <path_to_configuration>"
+if [ $# -ne 4 ]; then
+    echo "Usage: deploy-secrets.sh <namespace> <service-name> <path_to_configuration> <reference_of_decryption_barbican_secret>"
     exit 1
 fi
 
 ns=$1
 srv=$2
 conf=$3
+secretref=$4
 
     # cmsweb configuration area
     echo "+++ configuration: $conf"
     echo "+++ cms service : $srv"
     echo "+++ namespaces   : $ns"
+    echo "+++ secretref   : $secretref"
 
     if [ ! -d $conf/$srv ]; then
 	echo "Unable to locate $conf/$srv, please provide proper directory structure like <configuration>/<service>/<files>"
   	exit 1
     fi
+   
+    # Get SOPS decryption key and put it in config directory
+    mkdir -p ~/.config/sops/age/
+    rm ~/.config/sops/age/keys.txt
+    openstack secret get $secretref --payload_content_type application/octet-stream --file ~/.config/sops/age/keys.txt
```

Comment:
this command should be only executed if specific file is not present (but not deleted). If such file is present we should not execute this command.